src/target/cortexa.c

```diff
@@ -317,6 +330,54 @@ static size_t create_tdesc_cortex_a(char *buffer, size_t max_len)
 	return (size_t)total;
 }
 
+typedef struct bitfield_entry {
+	char *desc;
+	uint8_t bitnum;
+} bitfields_lut_s;
+
+static const bitfields_lut_s cortexa_dbg_dscr_lut[] = {
+	{"HALTED", 0U},
+	{"RESTARTED", 1U},
+	{"SDABORT_l", 6U},
+	{"ADABORT_l", 7U},
+	{"UND_l", 8U},
+	{"FS", 9U},
+	{"ITRen", 13U},
+	{"HDBGen", 14U},
+	{"MDBGen", 15U},
+	{"InstrCompl_l", 24U},
+	{"PipeAdv", 25U},
+	{"TXfull_l", 26U},
+	{"RXfull_l", 27U},
+	{"TXfull", 29U},
+	{"RXfull", 30U},
+};
+
+static void helper_print_bitfields(const uint32_t val, const bitfields_lut_s lut[], const size_t array_length)
```

Comment:
`const bitfields_lut_s lut[]` -> `const bitfields_lut_s *lut` please because C arrays do not pass like this so this code is unintentionally lying. It's thanks to how C arrays decay to pointers.

Reply:
Changed. I assumed I could use the ARRAY_LENGTH macro inside that function, but apparently passing arrays in C loses information about them. For now that's handled in the caller which passes the size alongside the array/pointer argument.

Reply:
Indeed, this is tied up with the same reason writing it as an array is an unintentional lie - the declaration becomes `const bitfields_lut_s *lut` before even the function declaration has finished being parsed. This means that by the time things like ARRAY_LENGTH() get to see it, you wind up asking for the length of a pointer to a struct divided by the length of that struct, rather than anything to do with arrays.